src/ApiClient/ServerClient.ts

```diff
@@ -69,14 +276,12 @@ export default new (class ServerClient extends ApiClient<IEvents> {
         //Why is this here? Because otherwise it creates an import loop, grrrrr
         configOptions.apipath.callback = (): void => {
             console.log("Reinitializing API");
-            this.initApi()
-                .then(() => {
-                    console.log("API Reinitialized");
-                })
-                .catch(() => {
-                    //The API failing to initialize is a big nono, start all over again.
-                    window.location.reload();
-                });
+            try {
+                this.initApi();
+                console.log("API Reinitialized");
+            } catch (error) {
+                window.location.reload();
+            }
```

Comment:
```suggestion
            window.location.reload();
```
This was buggy and never worked properly, please kill it with fire